js/main.js

```diff
@@ -1,128 +1,197 @@
+      toggleIcon($(this).attr("id"));
     });
   });
 
-
-/*
+  /*
   The boundaries for Nashville are
   upper-left: [35.9758, -86.9815]
   lower-right: [36.3350, -86.4542]
 
   This center and zoom doesn't capture the entire box, but includes
   all of the points in our datasets without a lot of blank space.
 */
-  var map = L.map('map').setView(
+  var map = L.map("map").setView(
     // center and zoom
-    [36.165818, -86.784245], 12
-  )
+    [36.165818, -86.784245],
+    12
+  );
 
   //varables for the map. points are latitude and longitude points
   //markers are leaflet markers
-  var firePoints=[], parksPoints=[], policePoints=[], wifiPoints=[], communityPoints=[],
-    fireMarkers=[], policeMarkers=[], wifiMarkers=[], parksMarkers=[], communityMarkers=[]
+  var firePoints = [],
+    parksPoints = [],
+    policePoints = [],
+    wifiPoints = [],
+    communityPoints = [],
+    fireMarkers = [],
+    policeMarkers = [],
+    wifiMarkers = [],
+    parksMarkers = [],
+    communityMarkers = [];
   //set up the icons for the markers
-  var fireIcon = new L.Icon({iconUrl:'../images/firestation-color.png',
-                            iconSize:[45,45]}),
-      policeIcon = new L.Icon({iconUrl:'../images/policestation-color.png',
-                              iconSize:[45,45]}),
-      wifiIcon = new L.Icon({iconUrl:'../images/wifi-color.png',iconSize:[45,45]}),
-      parksIcon = new L.Icon({iconUrl:'../images/park-color.png',iconSize:[45,45]}),
-      communityIcon = new L.Icon({iconUrl:'../images/home-color.png',iconSize:[45,45]});
+  var fireIcon = new L.Icon({
+    iconUrl: "../images/firestation-color.png",
+    iconSize: [45, 45]
+  }),
+    policeIcon = new L.Icon({
+      iconUrl: "../images/policestation-color.png",
+      iconSize: [45, 45]
+    }),
+    wifiIcon = new L.Icon({
+      iconUrl: "../images/wifi-color.png",
+      iconSize: [45, 45]
+    }),
+    parksIcon = new L.Icon({
+      iconUrl: "../images/park-color.png",
+      iconSize: [45, 45]
+    }),
+    communityIcon = new L.Icon({
+      iconUrl: "../images/home-color.png",
+      iconSize: [45, 45]
+    });
 
   //create the layer for the map from MapQuest
-  L.tileLayer('http://{s}.tile.osm.org/{z}/{x}/{y}.png', {
-    attribution: '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors',
-}).addTo( map );
+  L.tileLayer("http://{s}.tile.osm.org/{z}/{x}/{y}.png", {
+    attribution: '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
+  }).addTo(map);
 
-//Convert our JSON files into latitude longitude points in the "Points"
-//variables
-$.get('/datasets/community-centers-cleaned.json',function(data){
-  data.forEach(function(loc){
-    communityPoints.push([+loc.location[1],+loc.location[0]]);
+  function processJSON(data, locationKey) {
+    return data.data.map(function(el) {
+      var lat = el[locationKey][1];
+      var long = el[locationKey][2];
+      var location = ["", ""];
```

Comment:
yes look at this data set https://data.nashville.gov/api/views/74d7-b74t/rows.json.